### DISCUSSION TITLE
Seeking help with ROI for MultiChoise scan

### DISCUSSION THREAD

**Discussion by nyb12** [Sunbird-Saral/Community#72] [Q&A]
Hello everyone,

- We are currently facing an issue with scanning OMR sheets. Despite creating an ROI for a specific type of OMR sheet (any_1s_5q_multiChoise) with the assistance of Manoj Nath, the ROI does not match the exam details.
- The problem occurring is that the OMR is getting scanned but the sdk i.e(scanning by camera) ,but gets stuck on the scanning screen and doesn't return the values and also remains on the same page.

- other ORM are getting scanned we tried this example :

[assam_10s_10quesions_7digitsid_omr.pdf](https://github.com/Sunbird-Saral/Community/files/11272391/assam_10s_10quesions_7digitsid_omr.pdf)

OMR we are trying to scan :

- I have also attached the page which we are trying to scan.
- and Layout which we have used

Thank you in advance for your time and assistance.

![primarydata](https://user-images.githubusercontent.com/127941199/233057525-50af102f-62c2-4cdc-b4d3-56f5dcdf6c97.PNG)
[layout.docx](https://github.com/Sunbird-Saral/Community/files/11272369/layout.docx)

**Comment by sagarihanda** [comment]
@harish-challa Please help here.

**Comment by harish-challa** [comment]
@nyb12 Possible reasons why its stuck on the scanning screen – follow up questions 

1. roi json provided in layout.docx belongs to which layout? As there are two images - one is pdf and other one is coloured image.  
2. roi.json (any_1s_5q_multiChoice) provided in layout.docx is not appropriate. 
3. No student Id present on coloured image. 
4. coloured image is not proper, it should be on A4 size paper and should follow layout Design Guidelines [Layout Design Guidelines](https://saral.sunbird.org/use/layout-design-guidelines) 
5. We tried to reproduce issue on our end, for us also it is stuck on the scanning screen. For coloured image it is not even detecting black dots but for other pdf it is detecting that four dots.  
6. For pdf, roi.json is not correct that’s the reason it's happening. 
7. For coloured image, layout guidelines are not followed. 
8. Are you using Saral reference app or Saral SDK with your own custom application?  

COMMENTS: - Please check on all above steps.

**Reply by manjirisulekavadi** [reply]
@harish-challa most of the guidelines followed by the team now. 
[MultiplePagesPerStudent-BlankTemplate.pdf](https://github.com/Sunbird-Saral/Community/files/11321257/MultiplePagesPerStudent-BlankTemplate.pdf)
[BlankTemplate.pdf](https://github.com/Sunbird-Saral/Community/files/11321259/BlankTemplate.pdf)
It's still not working. Any other aspects are missed. The use case is we need (1) multiple pages per student id, need blocks for color pictures and the OMR bubble 1 and per cell

**Reply by sagarihanda** [reply]
Could have a brief call to discuss these issues tomorrow? @harish-challa

**Reply by sagarihanda** [reply]
Would also like to understand the overall use case and context of adoption further.

**Reply by sagarihanda** [reply]
@Shreyashi

**Reply by manjirisulekavadi** [reply]
Updates - got some success with one page custom format.  Trying with the multiple pages per student format. (Will share the templates tomorrow) These are dummy templates. Will check with the multi-color designed format (with ROI related formatting guidelines complied) and update. - temporarily we can assume that problem is resolved. Will share the dummy templates for this use case tomorrow and can discuss the use case for details.

**Comment by nyb12** [comment]
hi ,

- using Saral reference app with branch as [v1.5.8-RC-up-apk]

- we created the form as per layout design guidelines of saral which i have attached below ,so the problem occurring here is the same that it's getting stuck an capturing OMR ,
(This is for multipage scanning)

- pdf which we are scanning : [Student1MultiplePages.pdf](https://github.com/Sunbird-Saral/Community/files/11339392/Student1MultiplePages.pdf)

- this is the New ROI: [newROI.txt](https://github.com/Sunbird-Saral/Community/files/11339382/newROI.txt)

- Exam corresponding to ROI :[questions.txt](https://github.com/Sunbird-Saral/Community/files/11339387/questions.txt)

- the screenshot of how i am trying to map the ROI in Vott,

![Screenshot (10)](https://user-images.githubusercontent.com/127941199/234755689-73e424c2-1bd4-419b-90c6-71ce6fc54a30.png)


if I am missing any point or part while creating ROI / mapping tags /creating layout /creating exams, here please suggest.

Thank you in advance for your time and assistance.

**Reply by ShreyashiSoni** [reply]
@harish-challa Please go through these and suggest. 

We can also get on a quick call with the team to resolve the issues they are facing. Let me know @nyb12 @harish-challa

**Reply by harish-challa** [reply]
Hello @nyb12 . Please see our findings in the attached doc. 
[roiFindings.txt](https://github.com/Sunbird-Saral/Community/files/11342526/roiFindings.txt)

**Reply by nyb12** [reply]
Hi @harish-challa , @ShreyashiSoni ,we can take call tomorrow for this issue ,
and i also tried the new ORI as per suggested by @harish-challa sir,

[newROI.txt](https://github.com/Sunbird-Saral/Community/files/11345265/newROI.txt)
for the below OMR
[PrathamBlankTemplate2.pdf](https://github.com/Sunbird-Saral/Community/files/11345276/PrathamBlankTemplate2.pdf)

@apurvabisht97 Please need your guidance here ,
Thank you for your time and cooperation

**Reply by ShreyashiSoni** [reply]
Hi tomorrow is little difficult for me. 
I can do after 7pm if that works or you and Harish can go ahead and have a catch up call. 

Let me know if you want me to share invite to both of you. Thanks!

@harish-challa @nyb12

**Reply by manjirisulekavadi** [reply]
@harish-challa Possible to have a call with the team in the first half today.

**Reply by harish-challa** [reply]
@nyb12 @manjirisulekavadi We can connect at 12 Noon? . Sharing meet link for the same. 

ROI Discussion Problem Determination
Friday, 28 April · 12:00 – 1:00pm
Time zone: Asia/Kolkata
Google Meet joining info
Video call link: https://meet.google.com/vfh-soeu-utm

**Reply by manjirisulekavadi** [reply]
Thanks @harish-challa

**Reply by ShreyashiSoni** [reply]
Hi @nyb12 @harish-challa Is the issue resolved? Can you share a brief update. Thank You.

**Reply by nyb12** [reply]
hi @ShreyashiSoni  the issue is resolved regarding OMR scan.
Thank you.
